packages/sit-onyx/src/composables/useMoreList.ts

```diff
@@ -56,98 +44,113 @@ export type UseMoreListOptions = {
+ *     <div ref="listRef" class="more-list__elements">
+ *        <OnyxNavButton v-for="i in 16" ref="componentRefs" :key="i" :label="`Nav button ${i}`" />
+ *     </div>
+ *
+ *     <div ref="moreIndicatorRef" class="more-list__indicator">
+ *        +{{ more.hiddenElements.value.length }} more
+ *     </div>
  *   </div>
  * </template>
  *
  * <style lang="scss">
- * @use "../../styles/mixins/layers.scss";
+ * .more-list {
+ *   display: flex;
+ *   align-items: center;
+ *   gap: var(--onyx-spacing-4xs);
  *
- * .onyx-more {
- *   @include layers.component() {
- *     display: flex;
- *     align-items: center;
+ *   &__elements {
+ *     display: inherit;
+ *     align-items: inherit;
+ *     gap: inherit;
  *     overflow-x: clip;
+ *    }
+ *
+ *   &__indicator {
+ *     min-width: max-content;
+ *     max-width: 100%;
  *   }
  * }
  * </style>
  * ```
  */
 export const useMoreList = (options: UseMoreListOptions) => {
   const visibleElements = ref<string[]>([]);
+  const hiddenElements = ref<string[]>([]);
 
-  const hiddenElements = computed(() => {
-    return Array.from(options.componentRefs.keys()).filter(
-      (key) => !visibleElements.value.includes(key),
-    );
-  });
-
-  const observer = ref<IntersectionObserver>();
-  onBeforeUnmount(() => observer.value?.disconnect());
+  const { width: parentWidth } = useResizeObserver(options.parentRef);
+  const { width: moreIndicatorWidth } = useResizeObserver(options.moreIndicatorRef);
 
-  watch(
-    [options.parentRef, options.componentRefs, options.disabled],
-    () => {
-      observer.value?.disconnect(); // reset observer before all changes
+  // type casting is needed to prevent TypeScript from unwrapping the type which will lead to "width" being a single number instead of Ref<number>
+  const componentMap = ref(new Map()) as Ref<Map<string, { width: Ref<number> }>>;
 
-      const root = getTemplateRefElement(options.parentRef.value);
-      if (!root || options.disabled?.value) {
-        visibleElements.value = [];
-        return;
-      }
-
-      observer.value = new IntersectionObserver(
-        (changeEntries) => {
-          // changeEntries contains all changed component visibilities (not all available components)
-          // if component is shown, intersectionRatio is 1, otherwise its completely or partially hidden
-          const shownIds: string[] = [];
-          const hiddenIds: string[] = [];
+  /**
+   * Map of all component widths. Key = component ID, value = component width.
+   * If component is hidden, this map will still include the previous width.
+   */
+  const memorizedComponentWidthMap = ref(new Map<string, number>());
+
+  // sync widthMap to components but keep component width if it changes to zero (e.g. because the element is hidden)
+  // so we can continue calculating if hidden elements would fit into the available width if resized
+  watchEffect(() => {
+    componentMap.value.forEach((value, key) => {
```

Comment:
Imho we should move the caching to the `useMoreListChild`, e.g.:

```ts
  watchEffect(() => {
    if (newWidth) {
      moreContext.componentMap.value.set(id, { width: newWidth });
    }
  });
```

This reduces the complexity here, in `useMoreList`.